### PR TITLE
New remove journey middleware to set the page banner

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

None

### Change description

A PoC just to see if this will work. Looking good (remove banner displays, also after validation failures), although question around the changes required for unit tests.

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: test/controllers/secure.register.filter.controller.spec.ts [hunk 1]
```diff
@@ -43,6 +45,9 @@ const mockSetExtraData = setExtraData as jest.Mock;
 const mockServiceAvailabilityMiddleware = serviceAvailabilityMiddleware as jest.Mock;
 mockServiceAvailabilityMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
 
+const mockRemoveJourneyMiddleware = removeJourneyMiddleware as jest.Mock;
```
Comment (mwestacott) [inline test/controllers/secure.register.filter.controller.spec.ts:48]: Tested locally. Seems to work, but quite a few tests fail.

Would need to add this to each failing test or is there some way to define a general mock implementation across tests @markpit ?

Comment (markpit) [reply test/controllers/secure.register.filter.controller.spec.ts:48]: yes, you can create the mock in a separate file and import it

Comment (markpit) [reply test/controllers/secure.register.filter.controller.spec.ts:48]: eg https://github.com/companieshouse/confirmation-statement-web/blob/main/test/controllers/create.transaction.controller.unit.ts#L5

Comment (mwestacott) [reply test/controllers/secure.register.filter.controller.spec.ts:48]: Thanks. Looks simple enough. But we would then need to add the import statement to every (failing) test file, yes?

Comment (markpit) [reply test/controllers/secure.register.filter.controller.spec.ts:48]: yes, you'd still need to import it

File: src/middleware/navigation/remove/remove.journey.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import {
+  JourneyType
+} from "../../../config";
+import { logger } from "../../../utils/logger";
+import { isRemoveJourney } from "../../../utils/url";
+
+export const removeJourneyMiddleware = (req: Request, res: Response, next: NextFunction) => {
+
+  if (isRemoveJourney(req)) {
```
Comment (mattch1) [inline src/middleware/navigation/remove/remove.journey.middleware.ts:15]: Good idea this makes the remove test more explicit, are there plans to delete the existing isRemoveJourney from url.ts?

Comment (mwestacott) [reply src/middleware/navigation/remove/remove.journey.middleware.ts:15]: No, that function will still be needed as called in a number of places (mainly controller code).

**Pull request merged**